File: final-array-state-after-k-multiplication-operations-i.ts
{
    function getFinalState(nums: number[], k: number, multiplier: number): number[] {
        while(k) {
            let index_min = 0;
            let min = nums[0];

            for (let i = 1; i < nums.length; i++) {
                if (nums[i] < min) {
                    index_min = i;
                    min = nums[i];
                }
            };
            nums[index_min] *= multiplier;
            k--;
        }

        return nums;
    };

    console.log(getFinalState([2,1,3,5,6], 5, 2))
}

File: final-array-state-after-k-multiplication-operations-i.rs
pub fn get_final_state(mut nums: Vec<i32>, mut k: i32, multiplier: i32) -> Vec<i32> {
    while k != 0 {
        let mut index_min = 0;
        let mut min = nums[index_min];

        for i in 1..nums.len() {
            if nums[i] < min {
                index_min = i;
                min = nums[i];
            }
        }
        nums[index_min] *= multiplier;
        
        k -= 1;
    }

    nums
}